test/compilable/test19778.d

```diff
@@ -0,0 +1,6 @@
+struct S
+{
+    int[] data;
+}
+immutable X = S([]);
+enum len = X.data.length;
```

Comment:
Interestingly, this produces a compile-time error in current DMD:  https://run.dlang.io/is/upkuqa

Should this be ported to the DDMD frontend?